inconsistent test results

<!--
Instructions:
 * Fill out the sections below, replace …'s with information about your issue
 * Use the 'preview' function above this text box to verify formatting before submitting
-->

### Observed behavior
<!--
Description of the behavior that was observed, including screenshots or other references when applicable
-->

I noticed that in Travis, the results of tests `test_deleted_resources`, `test_new_resources` and `test_update_resources` are inconsistent. They would fail sometimes, but after restarting the jobs, they would pass.

### Expected behavior
<!--
Description of what behavior was expected but did not occur
-->

test results should be consistent

### User-facing consequences
<!--
Implications and real-world consequences for learners, coaches, admins, and other users of the application
-->

developers need to restart Travis jobs to make the tests pass

### Errors and logs
<!--
Relevant logs from:
 * the command line
 * ~/.kolibri/logs/kolibri.txt
 * the browser console

Please wrap errors in triple backticks for clean formatting like this:
```
01:10 info: something happened
01:12 error: something bad happened
```
-->

```
_________________ ChannelMixedTestCase.test_deleted_resources __________________
1387
1388
self = <kolibri.core.content.test.test_channel_upgrade.ChannelMixedTestCase testMethod=test_deleted_resources>
1389
1390
    def test_deleted_resources(self):
1391
        resources_to_be_deleted_count = count_removed_resources(
1392
            self.content_db_path, self.channel_id
1393
        )
1394
    
1395
>       self.assertEqual(resources_to_be_deleted_count, 5)
1396
E       AssertionError: 4 != 5
1397
1398
kolibri/core/content/test/test_channel_upgrade.py:591: AssertionError
1399

___________________ ChannelMixedTestCase.test_new_resources ____________________
1402
1403
self = <kolibri.core.content.test.test_channel_upgrade.ChannelMixedTestCase testMethod=test_new_resources>
1404
1405
    def test_new_resources(self):
1406
        (
1407
            new_resource_ids,
1408
            new_resource_content_ids,
1409
            new_resource_total_size,
1410
        ) = get_new_resources_available_for_import(
1411
            self.content_db_path, self.channel_id
1412
        )
1413
    
1414
        self.assertEqual(
1415
            set(new_resource_ids),
1416
>           set(map(lambda x: x["id"], self.upgraded_channel.new_resources)),
1417
        )
1418
E       AssertionError: Items in the second set but not the first:
1419
E       '911f0394683140af9ffe0e13f66033a7'
1420
1421
kolibri/core/content/test/test_channel_upgrade.py:570: AssertionError

__________________ ChannelMixedTestCase.test_update_resources __________________
self = <kolibri.core.content.test.test_channel_upgrade.ChannelMixedTestCase testMethod=test_update_resources>
    def test_update_resources(self):
        (
            updated_resource_ids,
            updated_resource_content_ids,
            updated_resource_total_size,
        ) = get_automatically_updated_resources(self.content_db_path, self.channel_id)
    
        self.assertEqual(
            set(updated_resource_ids),
>           set(map(lambda x: x["id"], self.upgraded_channel.updated_resources)),
        )
E       AssertionError: Items in the second set but not the first:
E       'f92e802f5c1a4e2da12577a76c81a173'
kolibri/core/content/test/test_channel_upgrade.py:602: AssertionError
```

### Steps to reproduce
<!--
Precise steps that someone else can follow in order to see this behavior
-->

not sure. this seems to happen randomly in Travis

### Context
<!--
Tell us about your environment, including:
 * Kolibri version
 * Operating system
 * Browser
-->

Kolibri version: release-v0.14.x

--- a/kolibri/core/content/test/test_channel_upgrade.py
+++ b/kolibri/core/content/test/test_channel_upgrade.py
@@ -55,6 +55,8 @@ class ChannelBuilder(object):
     def __init__(self, levels=3):
         self.levels = levels
 
+        self.modified = set()
+
         try:
             self.load_data()
         except KeyError:
@@ -208,24 +210,30 @@ class ChannelBuilder(object):
     def duplicate_resources(self, num_resources):
         self.duplicated_resources = []
         for i in range(0, num_resources):
-            parent = self.recurse_tree_until_leaf_container(self.root_node)
-            child = random.choice(parent["children"])
+            child = None
+            while child is None or child["id"] in self.modified:
+                parent = self.recurse_tree_until_leaf_container(self.root_node)
+                child = random.choice(parent["children"])
             duplicate = self.duplicate_resource(child)
             self.duplicated_resources.append(duplicate)
             parent["children"].append(duplicate)
+            self.modified.add(duplicate["id"])
         self.generate_nodes_from_root_node()
 
     def move_resources(self, num_resources):
         self.moved_resources = []
         self.deleted_resources = []
         for i in range(0, num_resources):
-            parent = self.recurse_tree_until_leaf_container(self.root_node)
-            child = random.choice(parent["children"])
+            child = None
+            while child is None or child["id"] in self.modified:
+                parent = self.recurse_tree_until_leaf_container(self.root_node)
+                child = random.choice(parent["children"])
             moved = self.duplicate_resource(child)
             self.moved_resources.append(moved)
             self.deleted_resources.append(child)
             parent["children"].pop(parent["children"].index(child))
             parent["children"].append(moved)
+            self.modified.add(moved["id"])
         self.generate_nodes_from_root_node()
 
     def upgrade(self, new_resources=0, updated_resources=0, deleted_resources=0):
@@ -239,19 +247,26 @@ class ChannelBuilder(object):
             # To emulate a common occurrence that produces edge cases
             # we also update the parent's thumbnail here
             self.updated_thumbnails.extend(self.update_thumbnail(parent))
+            self.modified.add(child["id"])
 
         self.updated_resources = []
         self.updated_resource_localfiles = []
         for i in range(0, updated_resources):
-            parent = self.recurse_tree_until_leaf_container(self.root_node)
-            child = random.choice(parent["children"])
+            child = None
+            while child is None or child["id"] in self.modified:
+                parent = self.recurse_tree_until_leaf_container(self.root_node)
+                child = random.choice(parent["children"])
             self.updated_resource_localfiles.extend(self.update_resource(child))
             self.updated_resources.append(child)
+            self.modified.add(child["id"])
 
         self.deleted_resources = []
         for i in range(0, deleted_resources):
-            parent = self.recurse_tree_until_leaf_container(self.root_node)
-            child_index = random.randint(0, len(parent["children"]) - 1)
+            child = None
+            while child is None or child["id"] in self.modified:
+                parent = self.recurse_tree_until_leaf_container(self.root_node)
+                child_index = random.randint(0, len(parent["children"]) - 1)
+                child = parent["children"][child_index]
             child = parent["children"].pop(child_index)
             self.delete_resource_files(child)
             self.deleted_resources.append(child)

